### DISCUSSION TITLE
VFD configuration

### DISCUSSION THREAD

**Discussion by tex1988** [devmfc/debian-on-amlogic#89] [Q&A] [answered]
Hi, devmfc and the community,
I appreciate your amazing work
I have one question. I am using Vontar X4 gigabit with your image and noticed that the VFD screen works but displays something incomprehensible. Can we configure it like on Armbian with openvfd or is there any way to use openvfd instead of the builtin vfd service?
For now on my Vontar X4 I am getting such messages in dmesg
```
tvbox:~#dmesg | grep fd6551
[    0.203533] tm16xx fd6551: DEVMFC: tm16xx driver v1.96 probe
[    0.203637] tm16xx fd6551: Get stb GPIO failed: -2, assuming two wire protocol
```

Added: this error doesn't affect anything but segment-mapping is wrong in meson-sc2-vontarx4-gbit.dtb and I can't figure out how to define the correct.
If it will be helpful here is the original android dts file
[single.dts.zip](https://github.com/user-attachments/files/16875681/single.dts.zip)

Thank you in advance

**Comment by devmfc** [comment]
Only the mapping is wrong.
You could try to figure the right mapping, see */usr/lib/tm16xx-display/tm16xx-map.sh* for some examples.

Original DTB will not be very helpful. 

If you find the right mapping, please let me know, then I will fix the DTB for the next release.

Of course you can also disable my driver by removing the fd6551 node from the DTB.

**Reply by tex1988** [reply]
Unfortunately, I can't manage to find the right mapping so I removed fd6551 from dtb and moved to openvfd driver

**Comment by Machete2048** [comment]
Hello there, 

First things first: many thanks for the nice work on providing such images for these boxes.

i came up here while looking for some informations for this tm16xx display (i own a Vontar-X4 Gigabit too), 
and after playing around, here is the setting that worked for me : 

- change the following line in file /usr/lib/tm16xx-display/tm16xx-map.sh : 
```
echo ${map6[@]} > $dev_path/mapping
```
with: 
```
echo ${map2[@]} > $dev_path/mapping
```
- OPTIONAL: you can uncomment the tests and the end of the file. 
- chmod +x /usr/lib/tm16xx-display/tm16xx-map.sh
- execute   /usr/lib/tm16xx-display/tm16xx-map.sh  (to initiliaze mapping)

Cheers,

**Reply by devmfc** [reply]
Thanx for letting me know! 

Attached is a new DTB with hopefully the right mapping, so it will have the right setting from the start. Let me know if it works and I will include it in the next release.

[meson-sc2-vontarx4-gbit.zip](https://github.com/user-attachments/files/18259665/meson-sc2-vontarx4-gbit.zip)

**Reply by Machete2048** [reply]
Hi again, 

Thanks for this dtb. 
i've just tried it, and it seems to work :)

FYI, here's my  test procedure (running with 6.12.5 kernel) : 
- copied the provided dtb file to /boot/dtb-6.12.5-meson64/amlogic/meson-sc2-vontarx4-gbit-tm16.dtb 
- updated /boot/box-config/box-vontarx4_gigabit.config, and point "dtb_img" to the forementioned file.
- reboot
- profit 

btw, happy new year !

**Reply by devmfc** [reply]
Thannx! And great to hear, I will add it to the next release.

**Comment by jenshenk** [comment]
I have a more fine-tuning question if that is okay. But first of all thanks so much for this project - I keep being amazed how flawless everything works on my s905x3 boxes, I've the first running directly from emmc since yesterday :)

My question: How can I configure what information is being displayed on the VFD? I can't seem to find what is controlling the default rotating info. Thanks in advance!

**Reply by Machete2048** [reply, marked as answer]
Hi there, 
just look at the following file: /usr/lib/tm16xx-display/tm16xx-map.sh
it's full of test examples which should help you understand how it works, it's pretty straightforward :)

Cheers,

edit: the default LED display is controlled by a systemd service called "display-service" (which calls a simple bash script). important: don't forget to stop it when you want to do some tests..

**Reply by jenshenk** [reply]
Thanks for pointing me in the right direction, in the other .sh file in the directory I at least finally found out that the number is the last part of the ip :) And a lot of thanks for the additional info I just saw... initially I tried playing around with it and nothing changed but stopping the service is a great tip ;)

**Comment by Deci8BelioS** [comment]
Hi, I made a modification of the display_service.sh script for **VONTAR X4** to enable wifi or ethernet, since in **VONTAR X4** there is no way (as far as I know) to get the cpu temperature it does not bring to display the temperature in case someone wants to try it.

[display_service.zip](https://github.com/user-attachments/files/18436041/display_service.zip)

- Display the time 

- wifi if connected or ethernet

- if usb flash drive is connected (not always does)

**the script goes in**: "\usr\lib\tm16xx-display"
